Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1032,61 +1032,106 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, unsigned int dstFo
 		}
 	}
 
-	// create the texture
-	TCacheEntryConfig config;
-	config.rendertarget = true;
-	config.width = scaled_tex_w;
-	config.height = scaled_tex_h;
-	config.layers = FramebufferManagerBase::GetEFBLayers();
+	u32 blockH = TexDecoder_GetBlockHeightInTexels(dstFormat);
+	const u32 blockW = TexDecoder_GetBlockWidthInTexels(dstFormat);
 
-	TCacheEntryBase* entry = AllocateTexture(config);
+	// Round up source height to multiple of block size
+	u32 actualHeight = ROUND_UP(tex_h, blockH);
+	const u32 actualWidth = ROUND_UP(tex_w, blockW);
 
-	entry->SetGeneralParameters(dstAddr, 0, dstFormat);
-	entry->SetDimensions(tex_w, tex_h, 1);
+	u32 num_blocks_y = actualHeight / blockH;
+	const u32 num_blocks_x = actualWidth / blockW;
 
-	entry->frameCount = FRAMECOUNT_INVALID;
-	entry->SetEfbCopy(dstStride);
-	entry->is_custom_tex = false;
+	// RGBA takes two cache lines per block; all others take one
+	const u32 bytes_per_block = dstFormat == GX_TF_RGBA8 ? 64 : 32;
 
-	entry->FromRenderTarget(dst, dstFormat, dstStride, srcFormat, srcRect, isIntensity, scaleByHalf, cbufid, colmat);
+	u32 bytes_per_row = num_blocks_x * bytes_per_block;
 
-	u64 hash = entry->CalculateHash();
-	entry->SetHashes(hash, hash);
+	bool copy_to_ram = !g_ActiveConfig.bSkipEFBCopyToRam;
+	bool copy_to_vram = true;
+
+	if (copy_to_ram)
+	{
+		g_texture_cache->CopyEFB(
+			dst,
+			dstFormat,
+			tex_w,
+			bytes_per_row,
+			num_blocks_y,
+			dstStride,
+			srcFormat,
+			srcRect,
+			isIntensity,
+			scaleByHalf);
+	}
+	else
+	{
+		u8* ptr = dst;
+		for (u32 i = 0; i < num_blocks_y; i++)
+		{
+			memset(ptr, 0, bytes_per_row);
+			ptr += dstStride;
+		}
+	}
+
+	if (g_bRecordFifoData)
+	{
+		// Mark the memory behind this efb copy as dynamicly generated for the Fifo log
+		u32 address = dstAddr;
+		for (u32 i = 0; i < num_blocks_y; i++)
+		{
+			FifoRecorder::GetInstance().UseMemory(address, bytes_per_row, MemoryUpdate::TEXTURE_MAP, true);
+			address += dstStride;
+		}
+	}
 
 	// Invalidate all textures that overlap the range of our efb copy.
 	// Unless our efb copy has a weird stride, then we want avoid invalidating textures which
 	// we might be able to do a partial texture update on.
-	if (entry->memory_stride == entry->BytesPerRow())
+	if (dstStride == bytes_per_row || !copy_to_vram)
 	{
 		TexCache::iterator iter = textures_by_address.begin();
 		while (iter != textures_by_address.end())
 		{
-			if (iter->second->OverlapsMemoryRange(dstAddr, entry->size_in_bytes))
-				iter = FreeTexture(iter);
-			else
+			if (iter->second->addr + iter->second->size_in_bytes <= dstAddr || iter->second->addr >= dstAddr /* + size_in_bytes*/ )
```

Comment:
If it's 0, there's no need to comment it out. So which PR did kill the size? Before phire's PRs, the size was returned on g_encoder->Encode for D3D and in TextureConverter::EncodeToRamFromTexture for OpenGL. But it could have been some other PR as well that killed it.

So why not calculate the size for efb copies? There have been issues in the past when the size for efb copies was not correct. As long as it has the size attribute, everybody expects it to have the value of the size. If it's not calculated, remove the attribute. That's the 2 choices i see here.